qiskit/backends/_backendutils.py

```diff
@@ -55,6 +55,22 @@ def discover_sdk_backends(directory=os.path.dirname(__file__)):
                         # Ignore backends that could not be initialized.
                         pass
 
+def discover_api_backends(api):
+    """Discover backends available on the Quantum Experience
+
+    Args:
+        api (IBMQuantumExperience): Quantum Experience API
+    """
+    from ._qeremote import QeRemote
+    QeRemote._api = api
```

Comment:
I was hoping to avoid passing it as an argument to the class constructor so I can instantiate remote and local classes identically. I think it would simplify the code since currently we have some local and remote branches/functions which might not be necessary. One change it is looking like would be needed to do to make this happen is to pass QuantumJob instances around in JobProcessor instead of qobj. I can then put job related attributes (e.g. wait, timeout, api?) in QuantumJob and the backend class can take out what it needs.